pkg/provider/azure_backoff.go

```diff
@@ -411,17 +411,33 @@ func (az *Cloud) CreateOrUpdatePIP(service *v1.Service, pipResourceGroup string,
 
 	rerr := az.PublicIPAddressesClient.CreateOrUpdate(ctx, pipResourceGroup, to.String(pip.Name), pip)
 	klog.V(10).Infof("PublicIPAddressesClient.CreateOrUpdate(%s, %s): end", pipResourceGroup, to.String(pip.Name))
-	if rerr != nil {
-		pipJSON, _ := json.Marshal(pip)
-		klog.Warningf("PublicIPAddressesClient.CreateOrUpdate(%s, %s) failed: %s, PublicIP request: %s", pipResourceGroup, to.String(pip.Name), rerr.Error().Error(), string(pipJSON))
-		az.Event(service, v1.EventTypeWarning, "CreateOrUpdatePublicIPAddress", rerr.Error().Error())
-		return rerr.Error()
+	if rerr == nil {
+		// Invalidate the cache right after updating
+		_ = az.pipCache.Delete(az.getPIPCacheKey(pipResourceGroup, to.String(pip.Name)))
+		return nil
 	}
 
-	return nil
+	pipJSON, _ := json.Marshal(pip)
+	klog.Warningf("PublicIPAddressesClient.CreateOrUpdate(%s, %s) failed: %s, PublicIP request: %s", pipResourceGroup, to.String(pip.Name), rerr.Error().Error(), string(pipJSON))
```

Comment:
we don't have to use rerr.Error().Error() here as the retry.Error has implemented String() method.

Reply:
This piece is everywhere in the file. `rerr` is not a `retryErr`, it's `*Error()`. First `Error()` converts it to `error` (in azure_error.go) and then err String() is printed.

Reply:
This format works, but if you look for simplicity, you could try to use rerr directly.

Reply:
sorry but a little bit confused... could you share how to simplify it?

Reply:
%w works well with error object.

Reply:
klong.Warningf("%s", rerr)